### PR TITLE
feat: add cloud account integration aws agentless scanning

### PR DESCRIPTION
**Pull request by dmurray-lacework**
Signed-off-by: Darren Murray <darren.murray@lacework.net>

## Summary

Add new cloud account for aws agentless scanning

## How did you test this change?

Run `api/_examples/cloud-accounts/aws-agentless-scanning/main.go`

## Issue

https://lacework.atlassian.net/browse/ALLY-1100

### CODE REVIEW

File: api/cloud_accounts.go [hunk 1]
```diff
@@ -98,17 +99,18 @@ const (
...
-	AzureAlSeqCloudAccount:    "AzureAlSeq",
-	AzureCfgCloudAccount:      "AzureCfg",
-	GcpAtSesCloudAccount:      "GcpAtSes",
-	GcpCfgCloudAccount:        "GcpCfg",
-	GcpGkeAuditCloudAccount:   "GcpGkeAudit",
+	NoneCloudAccount:                 "None",
+	AwsCfgCloudAccount:               "AwsCfg",
+	AwsCtSqsCloudAccount:             "AwsCtSqs",
+	AwsEksAuditCloudAccount:          "AwsEksAudit",
+	AwsAgentlessScanningCloudAccount: "AwsSidekick",
```
Comment (afiune) [inline api/cloud_accounts.go:106]:
```suggestion
	AwsSidekickCloudAccount: "AwsSidekick",
```

Comment (dmurray-lacework) [reply api/cloud_accounts.go]: done

File: api/cloud_accounts_aws_agentless_scanning.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package api
+
+// GetAwsAgentlessScanning gets a single AwsAgentlessScanning integration matching the provided integration guid
+func (svc *CloudAccountsService) GetAwsAgentlessScanning(guid string) (
```
Comment (afiune) [inline api/cloud_accounts_aws_agentless_scanning.go:22]: I'm sorry but we need to follow naming convention for the CloudAccount type and function(s)
```suggestion
func (svc *CloudAccountsService) GetAwsAwsSidekick(guid string) (
```

File: api/cloud_accounts_aws_agentless_scanning.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	response AwsAgentlessScanningResponse,
+	err error,
+) {
+	err = svc.get(guid, &response)
+	return
+}
+
+// UpdateAwsAgentlessScanning updates a single AwsAgentlessScanning integration on the Lacework Server
+func (svc *CloudAccountsService) UpdateAwsAgentlessScanning(data CloudAccount) (
```
Comment (afiune) [inline api/cloud_accounts_aws_agentless_scanning.go:31]:
```suggestion
func (svc *CloudAccountsService) UpdateAwsSidekick(data CloudAccount) (
```

File: api/cloud_accounts_aws_agentless_scanning.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+) {
+	err = svc.update(data.ID(), data, &response)
+	return
+}
+
+type AwsAgentlessScanningResponse struct {
+	Data AwsAgentlessScanning `json:"data"`
+}
+
+type AwsAgentlessScanning struct {
```
Comment (afiune) [inline api/cloud_accounts_aws_agentless_scanning.go:43]: I would encourage you to replace all occurances of `AgentlessScanning` in favor of `AwsSidekick`
 ```suggestion
type AwsSidekick struct {
```

**Review by afiune — CHANGES_REQUESTED** [review]

File: api/cloud_accounts_aws_agentless_scanning.go (hunk 4 above)
Comment (dmurray-lacework) [reply api/cloud_accounts_aws_agentless_scanning.go:43]: Done, however I do not like this naming convention.

**Review by afiune — APPROVED** [review]

**Pull request merged**